src/esp/bindings/AttributesBindings.cpp

```diff
@@ -43,9 +44,14 @@ void initAttributesBindings(py::module& m) {
   // NOLINTNEXTLINE(bugprone-unused-raii)
   py::class_<AbstractManagedObject, AbstractManagedObject::ptr>(
       m, "AbstractManagedObject");
+  // ==== AbstractFileBasedManagedObject ====
+  // NOLINTNEXTLINE(bugprone-unused-raii)
+  py::class_<AbstractFileBasedManagedObject, esp::core::AbstractManagedObject,
```

Comment:
Hold on. Are you sure don't want this inherit from the Python ```AbstractManagedObject``` binding? That should remove the need for NoLintNextLine too (it won't complain about unused raii if it's used in inheritance.)

Reply:
Ah, apparently this is a VALID way of binding the inheritance of PyBind11 as it will autoresolve, but if the class is already bound (which it is above), it's probably better to have the parent be the already PyBinded class. I think it will remove the need for the NoLint as an added bonus.

Reply:
disregard.